src/kOS.Safe/Compilation/Opcode.cs

```diff
@@ -679,7 +701,9 @@ public override void Execute(ICpu cpu)
             {
                 index = Convert.ToInt32(index);  // allow expressions like (1.0) to be indexes
             }
-            if (!(list is IIndexable)) throw new Exception(string.Format("Can't iterate on an object of type {0}", list.GetType()));
+
+            // Adjusted error message to reflect that there are now read-only indexable objects
+            if (!(list is IIndexable)) throw new Exception(string.Format("Can't set indexed elements on an object of type {0}", list.GetType()));
```

Comment:
I think this interferes with the new associtive array, which does allow string indeces.